### ISSUE TITLE
Can not use a parameter containing a comma to environment variables override

### ISSUE THREAD

**Issue by il-mtakada** [awslabs/aws-codebuild-jenkins-plugin#85]
The version of AWS CodeBuild Plugin is 0.41.

I tried to use a parameter for a build using parameterized build and tried to override the CodeBuild environment variable with this parameter.
However, when I specified a parameter containing a comma, a parsing error occurred.

The contents of the error are as follows.
```
[AWS CodeBuild Plugin] CodeBuild configured improperly in project settings
	> CodeBuild environment variable keys and values cannot be empty and the string must be of the form [{key, value}, {key2, value2}] (Service: null; Status Code: 0; Error Code: null; Request ID: null)
```

I think that the format of this part is better to use JSON.

For example.

```
[{"name":"key","value":"$VALUE"},{"name":"key2","value":"$VALUE2"}]
```

**Comment by leobaran-aws** [comment]
Thanks for letting us know about your use case--I'll look into this and see what we can do. It might not be possible to change the parsing strategy for the environment variables because of backward compatibility, but at the very least we should be able to support escaping characters.

**Comment by leobaran-aws** [comment]
Fixed in ede5949b18d36ce7a64bf1933d9965930d7158c3. For freestyle projects, you can now escape commas with one backslash:
```
[{VAR, value1\,value2}]
```

and for pipeline projects, escape with two backslashes:

```
[{VAR, value1\\,value2}]
```

This will be released in version `0.43` later today.
